nautobot/utilities/filters.py

```diff
@@ -370,11 +387,11 @@ def generate_query(self, filter_predicates, value):
             # to cast a value to an invalid type (e.g. `int("foo")` or `dict(42)`), ensure this
             # predicate is not included in the query.
             try:
-                value = preprocessor(value)
+                new_value = preprocessor(value)
```

Comment:
I realized that `value` was getting overwritten some cases, such as if it was stripped in one pass and not in another.